### ISSUE TITLE
bug: Experimental patch features

### ISSUE THREAD

**Issue by kenethc** [revanced/revanced-manager#634]
### Type

Other

### Bug description

I have Experimental Patches enabled in my settings. 

After every ReVanved Manager update, this setting gets disabled. 

So sometimes I patch the app and forget to enable the setting ... And this almost always cause the app to crash on open because of the patches i used while patching.

Please fix. 

### Steps to reproduce

1. Check setting for Experimental patches in Settings - should be enabled. 

2. Updating within the app doesn't work for me. So i usually download the update apk from GitHub and update as usual

3. After update, go to settings and check if Experimental patch is still enabled. Is always disabled for me. 

### Android version

Android 12 SKQ1

### Manager version

v52

### Target package name

YouTube 

### Target package version.

YouTube v17.39.47

### Installation type

Root

### Patches selected.

All except - 

Debugging
Micro-G 
Brightness and volume control

### Device logs (exported using Manager settings).

```shell
--------- beginning of system
01-09 14:15:59.747 18612 18612 I ForceDarkHelperStubImpl: initialize for app.revanced.manager.flutter , ForceDarkOrigin
01-09 14:15:59.911 18612 18612 D ViewScrollStubImpl: the value of ScrollFrictionRes is: 0.015
01-09 14:18:05.335 32042 32042 I ForceDarkHelperStubImpl: initialize for app.revanced.manager.flutter , ForceDarkOrigin
01-09 14:18:05.402 32042 32042 D ViewScrollStubImpl: the value of ScrollFrictionRes is: 0.015
01-09 14:18:14.764 32042 32042 W Looper  : PerfMonitor doFrame : time=1ms vsyncFrame=0 latency=1244ms procState=-1 historyMsgCount=115
--------- beginning of main
01-09 14:30:05.106 32042 32084 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:30:05.135 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x5102da, mIsProjectionMode = false
01-09 14:30:05.135 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:30:05.135 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:30:05.135 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:30:34.036 32042 32058 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:30:34.064 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0xef013bf, mIsProjectionMode = false
01-09 14:30:34.064 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:30:34.064 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:30:34.064 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:30:34.088 32042 32060 I OpenGLRenderer: Davey! duration=28940ms; Flags=1, FrameTimelineVsyncId=3029587, IntendedVsync=57898289809610, Vsync=57898289809610, InputEventId=810738555, HandleInputStart=57898290451244, AnimationStart=57898290451973, PerformTraversalsStart=57898290452129, DrawStart=57898299442546, FrameDeadline=57898306476276, FrameInterval=57898290433327, FrameStartTime=16666666, SyncQueued=57898299784890, SyncStart=57898299930202, IssueDrawCommandsStart=57898300209681, SwapBuffers=57898301482129, FrameCompleted=57927230080660, DequeueBufferDuration=642187, QueueBufferDuration=1217709, GpuCompleted=57927230080660, SwapBuffersCompleted=57898303114577, DisplayPresentTime=0, 
01-09 14:31:05.781 32042 32057 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:31:05.805 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x1883566, mIsProjectionMode = false
01-09 14:31:05.805 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:05.805 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:05.805 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:05.815 32042 32057 I OpenGLRenderer: Davey! duration=60680ms; Flags=1, FrameTimelineVsyncId=3029587, IntendedVsync=57898289809610, Vsync=57898289809610, InputEventId=810738555, HandleInputStart=57898290451244, AnimationStart=57898290451973, PerformTraversalsStart=57898290452129, DrawStart=57898299442546, FrameDeadline=57898323142942, FrameInterval=57898290433327, FrameStartTime=16666666, SyncQueued=57898299784890, SyncStart=57898299930202, IssueDrawCommandsStart=57898300209681, SwapBuffers=57898301482129, FrameCompleted=57958970732262, DequeueBufferDuration=642187, QueueBufferDuration=1217709, GpuCompleted=57958970732262, SwapBuffersCompleted=57898303114577, DisplayPresentTime=0, 
01-09 14:31:21.623 32042 32643 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:31:21.636 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x2c85969, mIsProjectionMode = false
01-09 14:31:21.636 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:21.636 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:21.636 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:21.648 32042 32084 I OpenGLRenderer: Davey! duration=76513ms; Flags=1, FrameTimelineVsyncId=3029587, IntendedVsync=57898289809610, Vsync=57898289809610, InputEventId=810738555, HandleInputStart=57898290451244, AnimationStart=57898290451973, PerformTraversalsStart=57898290452129, DrawStart=57898299442546, FrameDeadline=57898339809608, FrameInterval=57898290433327, FrameStartTime=16666666, SyncQueued=57898299784890, SyncStart=57898299930202, IssueDrawCommandsStart=57898300209681, SwapBuffers=57898301482129, FrameCompleted=57974803207777, DequeueBufferDuration=642187, QueueBufferDuration=1217709, GpuCompleted=57974803207777, SwapBuffersCompleted=57898303114577, DisplayPresentTime=0, 
01-09 14:31:52.490 32042 32084 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:31:52.523 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x3e408aa, mIsProjectionMode = false
01-09 14:31:52.523 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:52.523 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:52.523 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:31:52.542 32042 32643 I OpenGLRenderer: Davey! duration=78476ms; Flags=1, FrameTimelineVsyncId=3032481, IntendedVsync=57927218776088, Vsync=57927218776088, InputEventId=310566132, HandleInputStart=57927220456285, AnimationStart=57927220457431, PerformTraversalsStart=57927220457639, DrawStart=57927228192639, FrameDeadline=57927252109420, FrameInterval=57927220427587, FrameStartTime=16666666, SyncQueued=57927228283212, SyncStart=57927228327587, IssueDrawCommandsStart=57927228363576, SwapBuffers=57927229038472, FrameCompleted=58005694868494, DequeueBufferDuration=391875, QueueBufferDuration=669844, GpuCompleted=58005694868494, SwapBuffersCompleted=57927230112743, DisplayPresentTime=0, 
01-09 14:32:42.116 32042 32643 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:32:42.145 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x3162a62, mIsProjectionMode = false
01-09 14:32:42.145 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:32:42.145 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:32:42.145 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:32:42.153 32042 32643 I OpenGLRenderer: Davey! duration=49641ms; Flags=1, FrameTimelineVsyncId=3040181, IntendedVsync=58005668440695, Vsync=58005668440695, InputEventId=775099795, HandleInputStart=58005670187817, AnimationStart=58005670189015, PerformTraversalsStart=58005670189588, DrawStart=58005690898130, FrameDeadline=58005701774027, FrameInterval=58005670153651, FrameStartTime=16666666, SyncQueued=58005691008390, SyncStart=58005691046828, IssueDrawCommandsStart=58005691096046, SwapBuffers=58005693818390, FrameCompleted=58055310418944, DequeueBufferDuration=2244218, QueueBufferDuration=540104, GpuCompleted=58055310418944, SwapBuffersCompleted=58005695239015, DisplayPresentTime=0, 
01-09 14:32:45.096 32042 32042 E BLASTBufferQueue: [ViewRootImpl[MainActivity]#19](f:0,a:1) Applying pending transactions on dtor 1
01-09 14:33:06.304 32042 32643 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:33:06.324 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0xa76f6a, mIsProjectionMode = false
01-09 14:33:06.324 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:06.324 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:06.325 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:06.336 32042 32643 I OpenGLRenderer: Davey! duration=24196ms; Flags=1, FrameTimelineVsyncId=3044718, IntendedVsync=58055296526994, Vsync=58055296526994, InputEventId=590476543, HandleInputStart=58055297360715, AnimationStart=58055297361965, PerformTraversalsStart=58055297362538, DrawStart=58055308608580, FrameDeadline=58055329860326, FrameInterval=58055297328840, FrameStartTime=16666666, SyncQueued=58055308707850, SyncStart=58055308743319, IssueDrawCommandsStart=58055308810923, SwapBuffers=58055309728580, FrameCompleted=58079492962268, DequeueBufferDuration=611042, QueueBufferDuration=391823, GpuCompleted=58079492962268, SwapBuffersCompleted=58055310754361, DisplayPresentTime=0, 
01-09 14:33:16.292 32042 32643 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:33:16.323 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0xf865ad4, mIsProjectionMode = false
01-09 14:33:16.323 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:16.323 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:16.323 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:16.331 32042 32643 I OpenGLRenderer: Davey! duration=34192ms; Flags=1, FrameTimelineVsyncId=3044718, IntendedVsync=58055296526994, Vsync=58055296526994, InputEventId=590476543, HandleInputStart=58055297360715, AnimationStart=58055297361965, PerformTraversalsStart=58055297362538, DrawStart=58055308608580, FrameDeadline=58055346526992, FrameInterval=58055297328840, FrameStartTime=16666666, SyncQueued=58055308707850, SyncStart=58055308743319, IssueDrawCommandsStart=58055308810923, SwapBuffers=58055309728580, FrameCompleted=58089488958515, DequeueBufferDuration=611042, QueueBufferDuration=391823, GpuCompleted=58089488958515, SwapBuffersCompleted=58055310754361, DisplayPresentTime=0, 
01-09 14:33:19.993 32042 32042 E BLASTBufferQueue: [ViewRootImpl[MainActivity]#23](f:0,a:1) Applying pending transactions on dtor 1
01-09 14:33:31.641 32042 32084 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:33:31.655 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x6b88c9c, mIsProjectionMode = false
01-09 14:33:31.656 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:31.656 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:31.656 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:33:31.665 32042 32643 I OpenGLRenderer: Davey! duration=49525ms; Flags=1, FrameTimelineVsyncId=3044718, IntendedVsync=58055296526994, Vsync=58055296526994, InputEventId=590476543, HandleInputStart=58055297360715, AnimationStart=58055297361965, PerformTraversalsStart=58055297362538, DrawStart=58055308608580, FrameDeadline=58055363193658, FrameInterval=58055297328840, FrameStartTime=16666666, SyncQueued=58055308707850, SyncStart=58055308743319, IssueDrawCommandsStart=58055308810923, SwapBuffers=58055309728580, FrameCompleted=58104822284238, DequeueBufferDuration=611042, QueueBufferDuration=391823, GpuCompleted=58104822284238, SwapBuffersCompleted=58055310754361, DisplayPresentTime=0, 
01-09 14:34:01.560 32042 32058 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:34:01.589 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0xe68eaf6, mIsProjectionMode = false
01-09 14:34:01.589 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:01.589 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:01.589 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:01.601 32042 32643 I OpenGLRenderer: Davey! duration=55295ms; Flags=1, FrameTimelineVsyncId=3047428, IntendedVsync=58079460834527, Vsync=58079477501193, InputEventId=0, HandleInputStart=58079478209143, AnimationStart=58079478209977, PerformTraversalsStart=58079478210133, DrawStart=58079490333518, FrameDeadline=58079494167859, FrameInterval=58079478195445, FrameStartTime=16666666, SyncQueued=58079490845445, SyncStart=58079491043050, IssueDrawCommandsStart=58079491199300, SwapBuffers=58079492237008, FrameCompleted=58134756780789, DequeueBufferDuration=447084, QueueBufferDuration=491458, GpuCompleted=58134756780789, SwapBuffersCompleted=58079493417529, DisplayPresentTime=0, 
01-09 14:34:19.154 32042 32643 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:34:19.181 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x783d00, mIsProjectionMode = false
01-09 14:34:19.181 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:19.181 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:19.181 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:19.203 32042 32060 I OpenGLRenderer: Davey! duration=62877ms; Flags=1, FrameTimelineVsyncId=3048443, IntendedVsync=58089475277880, Vsync=58089475277880, InputEventId=995116235, HandleInputStart=58089476215494, AnimationStart=58089476216275, PerformTraversalsStart=58089476216483, DrawStart=58089487479504, FrameDeadline=58089508611212, FrameInterval=58089476198931, FrameStartTime=16666666, SyncQueued=58089487617160, SyncStart=58089487665754, IssueDrawCommandsStart=58089487737473, SwapBuffers=58089488424452, FrameCompleted=58152353211928, DequeueBufferDuration=428646, QueueBufferDuration=399062, GpuCompleted=58152353211928, SwapBuffersCompleted=58089489069087, DisplayPresentTime=0, 
01-09 14:34:59.606 32042 32057 D AppScoutStateMachine: 32042-ScoutStateMachinecreated
01-09 14:34:59.629 32042 32042 D SurfaceView: UPDATE Surface(name=SurfaceView[app.revanced.manager.flutter/app.revanced.manager.flutter.MainActivity])/@0x19391bc, mIsProjectionMode = false
01-09 14:34:59.629 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:59.629 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:59.629 32042 32042 D SurfaceControl: nativeSetScreenProjection ===> setScreenProjection
01-09 14:34:59.640 32042 32060 I OpenGLRenderer: Davey! duration=58052ms; Flags=1, FrameTimelineVsyncId=3052790, IntendedVsync=58134743649332, Vsync=58134743649332, InputEventId=987052357, HandleInputStart=58134745585581, AnimationStart=58134745587195, PerformTraversalsStart=58134745587768, DrawStart=58134754684956, FrameDeadline=58134776982664, FrameInterval=58134745550997, FrameStartTime=16666666, SyncQueued=58134754833601, SyncStart=58134754877351, IssueDrawCommandsStart=58134755017820, SwapBuffers=58134755953549, FrameCompleted=58192796199256, DequeueBufferDuration=441250, QueueBufferDuration=422291, GpuCompleted=58192796199256, SwapBuffersCompleted=58134756885685, DisplayPresentTime=0,
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] The issue is related solely to the ReVanced Manager

**Comment by kenethc** [comment]
I enabled both experimental patches support.

Will have to wait for an update to check if this sticks.

**Comment by TheGoldenSun** [comment]
I'll just leave this here. 

![Screenshot_20230112_040619](https://user-images.githubusercontent.com/104534608/212024740-09a20193-5dff-4063-9f78-3a68da292ef6.png)

**Comment by ILoveOpenSourceApplications** [comment]
This seems to be fixed as I just updated from 0.0.55 to 0.0.56 with both experimental patches enabled the options still being enabled in the updated version. Though I did update with the manager only and not by downloading the apk from GitHub.
